src/view/sidepanel/sidepanel.cpp

```diff
@@ -40,7 +42,19 @@ void View::SidePanel::SidePanel::addTab(QWidget *widget, const QString &icon,
 void View::SidePanel::SidePanel::removeTab(int index)
 {
 	title->removeTab(index);
-	stack->removeWidget(stack->widget(index));
+	auto tab = stack->widget(index);
+
+	// Find tab in lookup map and remove it
+	QMutableMapIterator<std::string, QWidget*> iterator(*titleToTabWidgetLookup);
+	while (iterator.hasNext()) {
```

Comment:
Tabs can [store data](https://doc.qt.io/qt-5/qtabbar.html#tabData)! Storing the artist ID in the tab itself avoids this.
I also prefer always keeping brackets on their own line for readability.

Reply:
That was a great pointer. I ended up using the property() instead, as I didn't have direct access to the QTabBar class (at least as far as I could tell).